mitodl/odl-video-service
# Created React pages for collection list and detail

#### What are the relevant tickets?
Closes #137 

#### What's this PR do?
Adds React pages for collection list and detail

#### How should this be manually tested?
Log in with a user that has a few collections, and at least one collection with videos. Click on the collection and video links to make sure you are taken to the correct pages.

#### Where should the reviewer start?
`static/js/containers/CollectionDetailPage.js` and `static/js/containers/CollectionListPage.js`

#### Any background context you want to provide?
- `CollectionListPage.js` is a placeholder of sorts. We're not exactly sure yet about what we're going to do with the collection list page, and we might get rid of it. In the meantime, it made more sense to create a barebones list page in React than it did to leave the old static page. This is also why there are no `CollectionListPage.js` tests (yet)
- I started working on pulling out some of the common UI code (dialog boxes, common page structure, etc.) into separate files, but the diff was quickly ballooning up. I saved those changes to a different branch and opened a new issue to complete that work (https://github.com/mitodl/odl-video-service/issues/175). **For now, the sandwich menu and the share/edit links in the collections pages are not functional.**
- There's an issue with the react router not successfully changing the selected collection when the URL changes. I'll ask @aliceriot about this, but in the meantime you just need to reload the collection detail page if it's not showing the correct one.

#### Screenshots (if appropriate)
<img width="758" alt="ss 2017-08-31 at 08 29 32" src="https://user-images.githubusercontent.com/14932219/29923189-9525e0c6-8e26-11e7-873a-d110818837b2.png">
